### PR TITLE
Build legacy provider for windows

### PR DESCRIPTION
**Pull request by h-vetinari**
Rebase #96 now that the fix for the expired certificate has been released in 3.0.4

Closes #96 
Closes #76

CC @hmaarrfk

### CODE REVIEW

**Comment by hmaarrfk** [comment]
Can you confirm that this fixes the issues we face in downstream packages?

**Comment by h-vetinari** [comment]
> Can you confirm that this fixes the issues we face in downstream package?


How should I be able to completely confirm that...? We know that cryptography needs the legacy provider of openssl 3.0. It's built by default on Unix but not windows (for reasons I don't claim to know). We should just build it IMO. The migration has stalled long enough as it is.

**Review by hmaarrfk — APPROVED** [review]

**Comment by h-vetinari** [comment]
@jakirkham @ocefpaf PTAL when you have some cycles. This is another attempt at unblocking the OpenSSL 3 migration that's currently stuck at cryptography.

File: recipe/bld.bat [hunk 1]
```diff
@@ -5,10 +5,12 @@ if "%ARCH%"=="32" (
...
 
 REM Configure step
-perl configure %OSSL_CONFIGURE% ^
-    --prefix=%LIBRARY_PREFIX% ^
+perl configure %OSSL_CONFIGURE%   ^
+    --prefix=%LIBRARY_PREFIX%     ^
     --openssldir=%LIBRARY_PREFIX% ^
-    shared ^
+    no-module                     ^
+    enable-legacy                 ^
```
Comment (ocefpaf) [inline recipe/bld.bat:12]: LGTM but I'd love if @isuruf could chime in as well. He has an eye for possible failures :-)

Comment (h-vetinari) [reply recipe/bld.bat:12]: Gentle ping @isuruf. Can explain the context around openssl 3 providers if desired.

Comment (ocefpaf) [reply recipe/bld.bat:12]: He is probably too busy with grad school. Let's merge and observe, if something bad happens we pull the package.

**Comment by h-vetinari** [comment]
This worked, as in cryptography can now load the DLL also on windows. :)

**Pull request merged**